identity/users_test.go

```diff
@@ -39,6 +42,111 @@ func MockIdentityClient() *identity.IdentityClient {
 	}
 }
 
+func TestAccUser_ChangeUserPassword(t *testing.T) {
+	userLogin := testutils.RandPrefixString("TestAccUser", 32)
+	userEmail := fmt.Sprintf("%s@example.com", userLogin)
+	userPassword := testutils.RandString(32)
+	userNewPassword := testutils.RandString(32)
+
+	// Holds newly created user.
+	var newUser *identity.User
+
+	testutils.AccTest(t, testutils.TestCase{
+		Steps: []testutils.Step{
+
+			&testutils.StepClient{
+				StateBagKey: "user",
+				CallFunc: func(config *triton.ClientConfig) (interface{}, error) {
+					return identity.NewClient(config)
+				},
+			},
+
+			&testutils.StepAPICall{
+				StateBagKey: "user",
+				CallFunc: func(client interface{}) (interface{}, error) {
+					c := client.(*identity.IdentityClient)
+					ctx := context.Background()
+					input := &identity.CreateUserInput{
+						Email:    userEmail,
+						Login:    userLogin,
+						Password: userPassword,
+					}
+					user, err := c.Users().Create(ctx, input)
+					newUser = user
+					return user, err
+				},
+				CleanupFunc: func(client interface{}, _ interface{}) {
+					c := client.(*identity.IdentityClient)
+					ctx := context.Background()
+					input := &identity.DeleteUserInput{
+						UserID: newUser.ID,
```

Comment:
You want to leverage the `stateBag` which is the second `interface{}` argument in this function signature (currently named `_`). Using a type assertion you should receive a `identity.User` as this value.

I believe you want something like the following with a known test user name (`const` or randomly generated) for easier sweeping.

```go
CleanupFunc: func(client interface{}, stateBag interface{}) {
	user, userOk := stateBag.(*identity.User)
	if !userOk {
		log.Println("expected state to be User")
		return
	}

	if user.Name != testUserName {
		log.Printf("expected user to be named %s: found %s\n",
			testUserName, user.Name)
		return
	}

	c := client.(*identity.IdentityClient)
	ctx := context.Background()
	input := &identity.DeleteUserInput{
		UserID: user.ID,
	}
	err := c.Users().Delete(ctx, input)
	if err != nil {
		log.Println("could not delete user", user.ID)
	}
},
```